pmd-java/etc/grammar/Java.jjt

```diff
@@ -3128,17 +3132,17 @@ void TryStatement() :
   [ FinallyStatement() ]
 }
 
-void ResourceSpecification() :
+void ResourceSpecification() #ResourceList:
 {}
 {
     {checkForBadTryWithResourcesUsage();}
 	"("
 	Resources()
-	(LOOKAHEAD(2) ";")?
+	(LOOKAHEAD(2) ";" {jjtThis.setTrailingSemi();})?
 	")"	
 }
 
-void Resources() :
+void Resources() #void:
```

Comment:
Do you think, it's worth to keep this a separate production or should we just merge it into ResourceList?

Reply:
It's indeed clearer to merge it. I didn't want to clutter the diff, but we should keep the grammar clean